kong/db/dao/workspaces.lua

```diff
@@ -18,4 +26,14 @@ function Workspaces:truncate()
 end
 
 
+function Workspaces:select_by_name(key, options)
+  if kong.configuration.database == "off" and key == "default" then
+    -- it should be a table, not a single string
+    return { id = lmdb.get(DECLARATIVE_DEFAULT_WORKSPACE_KEY) or DECLARATIVE_DEFAULT_WORKSPACE_ID, }
+  end
+
+  return self.super.select_by_name(self, key, options)
+end
```

Comment:
```suggestion
function Workspaces:select_by_name(key, options)
  if kong.configuration.database == "off" and key == "default" then
    -- it should be a table, not a single string
    return { id = lmdb.get(DECLARATIVE_DEFAULT_WORKSPACE_KEY)
  end

  return self.super.select_by_name(self, key, options)
end
```

As talked with @dndx @chronolaw , we'd better insert the DECLARATIVE_DEFAULT_WORKSPACE_ID into lmdb if dbless kong instance starts. Then we don't need to return a hardcode value here directly.

Left a comment here to block the PR, need to understand how default workspace(id) works in Kong master.